### DISCUSSION TITLE
Scripts to extract data from MOPAC calculations?

### DISCUSSION THREAD

**Discussion by icamps** [openmopac/mopac#93] [Q&A]
I am doing calculations with MOPAC.  As I have many systems (around 40), extracting the date manually, besides being tiring it can cause errors. I already worked with ASE, but the data it can extract is very limited.

My question is: **Are there available scripts (Bach, Python, etc.) to automatically extract data from MOPAC calculations?**

Note: I am thinking in extracting all (or most all) the useful data: heat of formation, molecular orbitals (w/, w/o spin polarization), energies, frequencies, thermochemistry data, forces, etc.

Regards,

Camps

**Comment by godotalgorithm** [comment]
MOPAC's testing process does parse output files (`tests/compare_output.py`), but not in a way that is useful for extracting specific properties right now.

This is one of the motivating reasons to develop a Python interface to MOPAC beyond some of these other interfaces like ASE - they don't expose a lot of data and/or features. Right now, the path of least resistance for output extraction is to rely on the AUX files (use the `AUX` keyword and examine `input.aux` that is produced by running `input.mop`), which is a highly machine-readable text file. Would parsing the AUX files be sufficient for your needs, or is there data that you are interested in that is missing from the AUX files?

If a Python interface parses the AUX output, then what do you see as a convenient format for further use in a Python workflow? Would it be sufficient to have a MOPAC output data object that simply assigns each block of data to a sensibly named float/vector/matrix, or do you have something else in mind?

**Reply by icamps** [reply]
Thank you. I will take a look at the AUX files.

**Comment by zfekete2** [comment]
@icamps Not everything, but to  to handle vibrational calculations (plus some thermochemistry) I have a bunch of scripts, mostly (g)awk. These are mostly useful for my own work, but I can show if you're interested.
